src/GADM.jl

```diff
@@ -52,86 +53,86 @@ function dataread(path)
 end
 
+"""
+    getdataset(country)
+
+Downloads and extracts dataset of the given country code
+"""
+function getdataset(country)
+    isvalidcode(country) || throw(ArgumentError("please provide standard ISO 3 country codes"))
+    data = country |> download |> dataread
+end
+
 """
     getlayer(data, level) 
 
 Get layer of the desired `level` from the `data`.
 """
 function getlayer(data, level)
     nlayers = ArchGDAL.nlayer(data)
-    for l=0:nlayers-1
+    for l = 0:nlayers - 1
         layer = ArchGDAL.getlayer(data, l)
         lname = ArchGDAL.getname(layer)
         llevel = last(split(lname, "_"))
         string(level) == llevel && return layer
     end
-    throw(ArgumentError("valid levels are 0...$(nlayers-1)"))
+    throw(ArgumentError("asked for level $(level), valid levels are 0-$(nlayers - 1)"))
 end
 
 """
-    get(country, subregions...)
+    get(country, subregions...; children=false)
+Returns a Tables.jl columntable for the requested region
+Geometry of the region(s) can be accessed with the key `geom`
+The geometries are GeoInterface compliant Polygons/MultiPolygons.
 
-Returns the MULTIPOLYGON data for the requested region.
-
-Input: ISO 3 country code, and further subdivisions.
+1. country: ISO 3166 Alpha 3 country code  
+2. subregions: Full official names in hierarchial order (provinces, districts, etc.)  
+3. children: When true, function returns two columntables -> parent, children.  
+Eg. when children is set true when querying just the country,
+second return parameter are the states/provinces.  
 
 ## Examples  
   
 ```julia
-get("IND")
-get("IND", "Uttar Pradesh")
-get("IND", "Uttar Pradesh", "Lucknow")
+# columntable of size 1, data of India's borders
+data = get("IND")
+# parent -> state data, children -> table of all districts inside Uttar Pradesh
+parent, children = get("IND", "Uttar Pradesh"; children=true)
 ```
 """
-function get(country, subregions...) 
-    isvalidcode(country) || throw(ArgumentError("please provide standard ISO 3 country codes"))
-
-    data = country |> download |> dataread
+function get(country, subregions...; children=false)
+    data = getdataset(country)
+    nlayers = ArchGDAL.nlayer(data)
 
-    function multipolygon(geom)
-       if GeoInterface.geotype(geom) == :Polygon
-        mp = ArchGDAL.createmultipolygon()
-        ArchGDAL.addgeom!(mp, geom)
-        return mp
-       else
-        return geom
-       end
+    function filterlayer(layer, key, value, all=false)
+        table = ArchGDAL.Table(layer)
+        filtered = []
+        for row in Tables.rows(table)
+            field = row[Symbol(key)]
+            if all || occursin(lowercase(value), lowercase(field))
+                push!(filtered, row)
+            end
+        end
+        return Tables.columntable(filtered)
     end
 
-    level = length(subregions)
-
-    # zoom into the appropriate layer for the query
-    layer = getlayer(data, level)
+    # p -> parent, is the requested region
```

Comment:
You don't need to change this, but I am just sharing so that you are aware. Notice that `p -> parent` is a valid syntax in Julia and that someone reading it will start thinking about anonymous functions. Here you just want to provide a helpful comment like: "find parent region"